### ISSUE TITLE
Allow custom configuration properties to be passed to Calcite

### ISSUE THREAD

**Issue by dheerajkulakarni** [trinodb/trino#7287]
`connection.getMetadata().storesUpperCaseIdentifiers()' returns true when called in DruidJdbcClient which has to be ideally false in case of Druid. This returns true because Avatica driver returns the properties of ORACLE properties by default, but at the moment Avatica JDBC driver doesn't allow to pass configuration like lex, caseSensitive etc [(https://calcite.apache.org/docs/adapter.html#jdbc-connect-string-parameters)](https://calcite.apache.org/docs/adapter.html#jdbc-connect-string-parameters) to calcite, which is making it impossible to set this kind of properties based on the type of JDBC driver. 

Changes to be considered in the code after this issue gets solved
`BaseJdbcClient.toRemoteSchema()` & `BaseJdbcClient.toRemoteTable()` methods from BaseJdbcClient has been overridden in DruidJdbcClient as `connection.getMetadata().storesUpperCaseIdentifiers()' was always returning true and tables and schemas are returned in upper case lettes when caseSensitiveMapping flag is off, so if this issue gets solved we can set properties in DruidJdbc Driver in a way that `connection.getMetadata().storesUpperCaseIdentifiers()` returns alse by default.

For more info refer to https://github.com/trinodb/trino/pull/7197

cc : @hashhar @findepi

**Comment by findepi** [comment]
Would it be possible for formulate this issue title in terms of benefits for the Druid connector?
The Avatica driver is not elsewhere, as far as i know.

**Comment by hashhar** [comment]
This should ideally be filed upstream (Druid or Apache Calcite) and we can use this to track what we need to do on our end once the driver gets fixed.

TL;DR The Avatica driver returns incorrect JdbcMetadata (unquoted identifier casing).
